src/nccl_ofi_topo.cpp

```diff
@@ -1455,20 +1455,26 @@ static int write_nic(hwloc_obj_t node, FILE *file, int indent)
 		}
 
 		NCCL_OFI_TRACE(NCCL_INIT, "GPU information: GPU speed \"%s\",  width \"%zu\"", pcie_gen[gpu_speed_idx], gpu_width);
-
 		/* In case we have multiple NICs in the group,
 		 * increase link speed if possible and decrease NIC
-		 * group size */
+		 * group size. Keep track of the remainder NICs when the number
+		 * of NICs is not a power of 2 */
+		size_t remainder_nics = 0;
 		while (group_size > 1 && speed_idx < gpu_speed_idx) {
 			++speed_idx;
+			if (group_size % 2 == 1) {
```

Comment:
I think we want to bump the gpu speed index instead of the lane index as far as possible.  The loop is going to suck to write, but I think you want to deal with remainder nics in here.

Reply:
Remember, it's ok if the PCIe link speed we report is *greater* than the speed of the combined nics, just not if it's *less* than the combined nics.  SO if you have 4 Gen4x8 NICs, reporting Gen5x16 or Gen6x8 is perfectly ok.  The net bandwidth number will still be group_size * ethernet speed, which is going to be the slow edge in the graph.

So I think the real right answer is to round up group_size to the next power of two before any of the loops and just not having all the remainder code.